core/src/sync/synchronization_protocol_handler.rs

```diff
@@ -817,6 +822,41 @@ impl SynchronizationProtocolHandler {
         send_message(io, peer, msg.as_ref(), SendQueuePriority::High)
     }
 
+    fn broadcast_status(&self, io: &NetworkContext) {
+        let best_info = self.graph.consensus.get_best_info();
+
+        let terminal_hashes = if let Some(x) = &best_info.terminal_block_hashes
```

Comment:
better to add a new method in BestInformation struct to avoid such duplicated code.